lang/types/value.go

```diff
@@ -65,17 +70,23 @@ func ValueOfGolang(i interface{}) (Value, error) {
 // a resource field. This is done with our "elvis" operator. When using this
 // function, if you pass in something with a nil value, then expect a panic or
 // an error if you're lucky.
-// FIXME: we should make sure we error instead of ever panic-ing on a nil value
-func ValueOf(v reflect.Value) (Value, error) {
-	value := v
+func ValueOf(value reflect.Value) (Value, error) {
+	// Gracefully handle invalid values instead of panic().
+	if !value.IsValid() {
+		return nil, &reflect.ValueError{Method: "types.ValueOf", Kind: reflect.Invalid}
```

Comment:
I am slightly afraid of returning an error from the reflect lib... Convince me it's okay, or change for a normal error please.

Reply:
Fixed